resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+
+U9. As a team member, I want to be able to update the progress of specific task (Not Started, In Progress, Complete).
+
+### 4. Stretch Goals
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+U11. Admin can set roles for a user to be a team member or manager. (Multiple roles)
+
+U12. As a team member, I want to be able to comment on a specific task I am assigned to.
+
+U13. As a manager, I want to be able to share and collaborate files and documents with my team.
+
+
+## 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+## 6. API
+
+### 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ProjectModel
+String projectId;
+String projectName;
+String projectStatus;
+List<Task> projectTasks;
+List<UserModel> projectTeamMembers;
+```
+
+```
+// TaskModel
+String taskId;
+String taskName;
+String taskDescription;
+String taskDueDate;
+String taskStatus;
+List<UserModel> taskAssignedUsers;
+Integer taskTimer;
+```
+
+```
+// UserModel
+String userId;
+String userName;
+String userRole;
+```
+
+
+### 6.3. _Create Project Endpoint_
+* Accepts `POST` requests to `/projects`
+* Accepts data to create a new Project with a given name, project status, and an optional list of tasks. Returns the new project, including a unique project ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```"projectName": "Project Name", ```<br>
+    ```"projectStatus": "Not Started", ```<br>
+    ```"projectStatus": "Not Started", ```<br>
+    ```"projectTeamMembers": ["userId_1", "userId_2"] ```<br>
+
+### 6.4. _Create Task Endpoint_
+* Accepts `POST` requests to `/tasks`
+* Accepts data to create a new Task with a given task name, description, due date, task status, and list of assigned team members. Returns the new Task, including a unique task ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```{"taskName": "Task Title",``` <br>
+    ```"taskDescription": "Task Description", ```<br>
+    ```"taskStatus": "Not Started", ```<br>
+    ```"taskAssignedUsers": ["userId_1", "userId_2"], ```<br>
+    ```"taskStatus": "open", ```<br>
+    ```"taskDueDate": "yyyy-mm-dd", ```<br>
+    ```"taskTimer": 60```<br>
+
+![](images/create-task-activity-SD.png)
+
+### 6.5. _Add Task To Project Endpoint_
```

Comment:
Your _Create Task Endpoint_ should include the project id when creating the task, so you wouldn't need this one (i.e. you shouldn't be able to create a task that's not part of a project).